### PR TITLE
added slippage to swap

### PR DESCRIPTION
**Pull request by mohsenParvar**
* added slippage to swap
* memoization for advanced options (change the advanced options and refresh the page, state will stay the same)
* fixed price fetch for tokens with the same key (all the tokens have their latest price, check the redux=> global=>tokenPrices)
* fixed the problem with swapping some of tokens (check with all the available tokens)
* added equivalent price for swapping tokens

### CODE REVIEW

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

File: src/abi/axialAggregator.json [hunk 1]
```diff
@@ -0,0 +1 @@
...":"renounceOwnership","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"_externalRouter","type":"address"}],"name":"setExternalRouter","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"_internalRouter","type":"address"}],"name":"setInternalRouter","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"components":[{"internalType":"uint256","name":"amountIn","type":"uint256"},{"internalType":"uint256","name":"amountOut","type":"uint256"},{"internalType":"address[]","name":"path","type":"address[]"},{"internalType":"address[]","name":"adapters","type":"address[]"}],"internalType":"struct IRouter.Trade","name":"_trade","type":"tuple"},{"internalType":"address","name":"_to","type":"address"},{"internalType":"uint256","name":"_fee","type":"uint256"},{"internalType":"bool","name":"_useInternalRouter","type":"bool"}],"name":"swap","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"newOwner","type":"address"}],"name":"transferOwnership","outputs":[],"stateMutability":"nonpayable","type":"function"}]
```
Comment (saito-sv) [inline src/abi/axialAggregator.json:1]: do we need this file? it seem that we already have it with `.ts`

File: src/app/index.tsx [hunk 2]
```diff
@@ -105,8 +106,8 @@ export function App() {
       />
       <Rewards pools={pools} />
       <Swap
-        swapRouterABI={SWAP_ROUTER_ABI}
-        swapRouterAddress={"0xBeD9dfE835cd2bB6775f344Ee5E3431b2CbF31FB"}
+        swapRouterABI={AGGREGATOR_ABI}
+        swapRouterAddress={"0x05F2F176b491ec11555cbE45B99748F999E13933"}
```
Comment (saito-sv) [inline src/app/index.tsx]: lets not hardcode this `address` please consider place it on a `const` with a proper name

**Review by bigwampa — DISMISSED** [review]

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-fix-mohsen-swap-slippage.dev-app.axial.exchange

**Review by saito-sv — APPROVED** [review]
LGTM

**Pull request merged**